packages/ethereum-contracts/scripts/deploy-test-environment.js

```diff
@@ -14,49 +14,49 @@ const deploySuperToken = require("./deploy-super-token");
  */
 module.exports = async function(
     callback,
-    { isTruffle, web3Provider, from } = {}
+    { isTruffle, web3, ethers, from } = {}
 ) {
     const errorHandler = err => {
         if (err) throw err;
     };
 
     try {
-        this.web3 = web3Provider ? new Web3(web3Provider) : web3;
-        if (!this.web3) throw new Error("No web3 is available");
+        validateWeb3Arguments({ web3, ethers, isTruffle });
 
         console.log("==== Deploying superfluid framework...");
         await deployFramework(errorHandler, {
-            isTruffle,
-            web3Provider: this.web3.currentProvider,
+            ethers,
+            web3,
             from
         });
         console.log("==== Superfluid framework deployed.");
 
         const tokens = ["fDAI", "fUSDC", "fTUSD"];
-        for (let i = 0; i < tokens.length; ++i) {
-            console.log(`==== Deploying test token ${tokens[i]}...`);
-            await deployTestToken(errorHandler, [":", tokens[i]], {
-                isTruffle,
-                web3Provider: this.web3.currentProvider,
-                from
-            });
-            console.log(`==== Test token ${tokens[i]} deployed.`);
+        await Promise.all([
```

Comment:
This we shouldn't do.

Transactions from the same account should not be parallized, it will create nonce conflict and will likely fail.

Reply:
Its currently not failing in truffle environment. Where would it fail? Only place I can think is if someone called it using MetaMask and had to manually sign each tx.

I'd prefer to keep it, if it reduces test time

Reply:
this will definitely fail for testnet/mainnet deployments.